packages/commonwealth/client/scripts/views/modals/manage_contract_template_modal.tsx

```diff
@@ -58,12 +59,42 @@ export class ManageContractTemplateModal extends ClassComponent {
     m.route.set(`/${scope}/new/contract_template`);
   }
 
-  handleConfirm(e) {
+  async handleConfirm(e) {
     e.preventDefault();
-    $(e.target).trigger('modalcomplete');
-    setTimeout(() => {
-      $(e.target).trigger('modalexit');
-    }, 0);
+
+    const communityId = app.activeChainId();
+    const { slug, displayOption, nickname, displayName } = this.form;
+
+    const communityTemplate = {
+      community_id: communityId,
+      // TODO do we know template id if we are about to create it
+      contract_id: 0,
+      // TODO get template id from parent
+      template_id: 0,
+    };
+
+    const communityTemplateMetadata = {
+      // TODO is cct_id the same as template_id ?
```

Comment:
No, Template is a separate entity. Think of template like the model that is scoped globally- all communities have access to them. CCTs always will correspond to a specific community.